server/courselist.rb

```diff
@@ -46,9 +47,23 @@ class CourseList < Sinatra::Base
   # where needed.
   set :latte_config, config_hash
 
-  # Location to find the configuration files.  This can't be overridden by values
-  # in the configuration file because this is what identifies the location
-  # of the file to read.
+  # print environment when debugging level.
+  ENV.each_pair do |key,value|
```

Comment:
This loop will execute whether or not debug logging is enabled.  It will access each env. variable and it may or may not log anything.  This is an unnecessary use of processing time.  You need to wrap the loop with a check for debug mode instead.